# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: extensions/spellcheck/src/mozSpellChecker.cpp
# Commit: 1c140849819b
# Full Hash: 1c140849819b044a9a288d531725bac43669a0c3
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2022-03-25 09:38:14
# Description:
#   Bug 1761085 - Check for existence of mEngine in mozSpellChecker; r=smaug
#   
#   With the changes to EditorSpellCheck::SetFallbackDictionary to use a
#   promise chain to support calling either SetCurrentDictionaryFromList or
#   SetCurrentDictionaries as required, it is now possible that
# ==============================================================================

diff -r 60beb289fe92 -r 1c140849819b extensions/spellcheck/src/mozSpellChecker.cpp
--- a/extensions/spellcheck/src/mozSpellChecker.cpp	Thu Mar 24 19:17:37 2022 +0000
+++ b/extensions/spellcheck/src/mozSpellChecker.cpp	Thu Mar 24 19:23:56 2022 +0000
@@ -466,6 +466,11 @@
 RefPtr<GenericPromise> mozSpellChecker::SetCurrentDictionaries(
     const nsTArray<nsCString>& aDictionaries) {
   if (XRE_IsContentProcess()) {
+    if (!mEngine) {
+      mCurrentDictionaries.Clear();
+      return GenericPromise::CreateAndReject(NS_ERROR_NOT_AVAILABLE, __func__);
+    }
+
     // mCurrentDictionaries will be set by RemoteSpellCheckEngineChild
     return mEngine->SetCurrentDictionaries(aDictionaries);
   }
@@ -518,6 +523,11 @@
   }
 
   if (XRE_IsContentProcess()) {
+    if (!mEngine) {
+      mCurrentDictionaries.Clear();
+      return GenericPromise::CreateAndReject(NS_ERROR_NOT_AVAILABLE, __func__);
+    }
+
     // mCurrentDictionaries will be set by RemoteSpellCheckEngineChild
     return mEngine->SetCurrentDictionaryFromList(aList);
   }
